p2p/exchange.go

```diff
@@ -121,30 +115,64 @@ func (ex *Exchange[H]) Head(ctx context.Context) (H, error) {
 		defer cancel()
 	}
 
+	reqParams := header.HeadParams{}
+	for _, opt := range opts {
+		opt(&reqParams)
+	}
+
+	var (
+		zero  H
+		peers = ex.trustedPeers()
+	)
```

Comment:
nit: the zero only needs to be allocated on error in the end, but its always allocated